src/main/java/gregtech/api/pipenet/PipeNetWalker.java

```diff
@@ -153,49 +164,51 @@ private boolean walk() {
             }
         }
 
-        return !isRunning() || walkers.size() == 0;
+        return !isRunning() || walkers.isEmpty();
     }
 
     private boolean checkPos() {
-        pipes.clear();
-        TileEntity thisPipe = world.getTileEntity(currentPos);
-        if (!(thisPipe instanceof IPipeTile<?, ?>)) {
-            GTLog.logger.fatal("PipeWalker expected a pipe, but found {} at {}", thisPipe, currentPos);
-            return false;
-        }
-        if (!getBasePipeClass().isAssignableFrom(thisPipe.getClass())) {
-            return false;
+        nextPipeFacings.clear();
+        nextPipes.clear();
+        if (currentPipe == null) {
+            TileEntity thisPipe = world.getTileEntity(currentPos);
+            if (!(thisPipe instanceof IPipeTile<?, ?>)) {
+                GTLog.logger.fatal("PipeWalker expected a pipe, but found {} at {}", thisPipe, currentPos);
+                return false;
+            }
+            if (!getBasePipeClass().isAssignableFrom(thisPipe.getClass())) {
+                return false;
+            }
+            currentPipe = (T) thisPipe;
```

Comment:
Should instanceof be checked here?

Reply:
What do you think the `isAssignableFrom()` method is for?